mobile/src/main/java/org/openhab/habdroid/ui/PreferencesActivity.kt

```diff
@@ -321,25 +324,25 @@ class PreferencesActivity : AbstractBaseActivity() {
             if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
                 Log.d(TAG, "Removing alarm clock pref")
                 preferenceScreen.removePreference(alarmClockPref)
-            } else {
-                updateAlarmClockPreferenceIcon(alarmClockPref,
-                    alarmClockPref.getPrefValue().toItemUpdatePrefValue())
-                updateAlarmClockPreferenceSummary(alarmClockPref,
-                    sendDeviceInfoPrefixPref.getPrefValue(),
-                    alarmClockPref.getPrefValue().toItemUpdatePrefValue())
-                alarmClockPref.setOnPreferenceChangeListener { preference, newValue ->
-                    val prefix = sendDeviceInfoPrefixPref.getPrefValue()
-                    @Suppress("UNCHECKED_CAST")
-                    val value = newValue as Pair<Boolean, String>
-                    updateAlarmClockPreferenceIcon(preference, newValue)
-                    updateAlarmClockPreferenceSummary(preference, prefix, value)
-                    true
+            }
+
+            phoneStatePref.setOnPreferenceChangeListener { preference, newValue ->
+                @Suppress("UNCHECKED_CAST")
+                val value = newValue as Pair<Boolean, String>
+                if (value.first && ContextCompat.checkSelfPermission(preference.context,
+                        Manifest.permission.READ_PHONE_STATE) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
Suggestion: Make an extension function: `fun Context.hasPermission(String permission): Boolean` for this.